libs/coin-modules/coin-ton/src/bridge/bridgeHelpers/txn.ts

```diff
@@ -0,0 +1,279 @@
+import { decodeAccountId, encodeTokenAccountId } from "@ledgerhq/coin-framework/account/index";
+import { encodeOperationId } from "@ledgerhq/coin-framework/operation";
+import { findTokenByAddressInCurrency } from "@ledgerhq/cryptoassets";
+import { Address } from "@ton/ton";
+import BigNumber from "bignumber.js";
+import { TonOperation } from "../../types";
+import { addressesAreEqual, isAddressValid } from "../../utils";
+import { fetchJettonTransactions, fetchTransactions } from "./api";
+import {
+  TonAddressBook,
+  TonJettonTransfer,
+  TonTransaction,
+  TonTransactionsList,
+} from "./api.types";
+
+export async function getTransactions(
+  addr: string,
+  startLt?: string,
+): Promise<TonTransactionsList> {
+  const txs = await fetchTransactions(addr, { startLt });
+  if (txs.transactions.length === 0) return txs;
+  let tmpTxs: TonTransactionsList;
+  let isUncompletedResult = true;
+
+  while (isUncompletedResult) {
+    const { lt, hash } = txs.transactions[txs.transactions.length - 1];
+    tmpTxs = await fetchTransactions(addr, { startLt, endLt: lt });
+    // we found the last transaction
+    if (tmpTxs.transactions.length === 1) {
+      isUncompletedResult = false;
```

Comment:
Hi! If the result is the same as last call, it will throw an error for the following condition, because the first transaction would be different to the last transaction: 

`if (hash !== tmpTxs.transactions[0].hash) throw Error("[ton] transaction hash does not match");`

On the other hand, if the result has one transaction, the loop will finish for the following condition: 
```
    // we found the last transaction
    if (tmpTxs.transactions.length === 1) {
      isUncompletedResult = false;
      break;
    }
```